pkg/reconciler/knativeserving/common/ingress_service.go

```diff
@@ -17,19 +17,45 @@ limitations under the License.
 package common
 
 import (
+	"fmt"
+	"strings"
+
 	mf "github.com/manifestival/manifestival"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"knative.dev/operator/pkg/apis/operator/v1alpha1"
 )
 
 // IngressServiceTransform pins the namespace to istio-system for the service named knative-local-gateway.
 // It also removes the OwnerReference to the operator, as they are in different namespaces, which is
 // invalid in Kubernetes 1.20+.
-func IngressServiceTransform() mf.Transformer {
+func IngressServiceTransform(ks *v1alpha1.KnativeServing) mf.Transformer {
 	return func(u *unstructured.Unstructured) error {
 		if u.GetAPIVersion() == "v1" && u.GetKind() == "Service" && u.GetName() == "knative-local-gateway" {
 			u.SetNamespace("istio-system")
 			u.SetOwnerReferences(nil)
+			config := ks.GetSpec().GetConfig()
+			if data, ok := config["istio"]; ok {
```

Comment:
What if the `config[“istio”]` is not present? Or could that be the case when different ingress is used?

Reply:
if neither `istio` nor `config-istio` exists, we will use the default ns `u.SetNamespace("istio-system")`